flytekit/common/tasks/pytorch_task.py

```diff
@@ -0,0 +1,121 @@
+            task_function=task_function,
+            task_type=task_type,
+            discovery_version=discovery_version,
+            retries=retries,
+            interruptible=interruptible,
+            deprecated=deprecated,
+            storage_request=instance_storage_request,
+            cpu_request=instance_cpu_request,
+            gpu_request=instance_gpu_request,
+            memory_request=instance_memory_request,
+            storage_limit=instance_storage_limit,
+            cpu_limit=instance_cpu_limit,
+            gpu_limit=instance_gpu_limit,
+            memory_limit=instance_memory_limit,
+            discoverable=discoverable,
+            timeout=timeout,
+            environment=environment,
+            custom=_MessageToDict(pytorch_job)
+        )
+
+    @_exception_scopes.system_entry_point
+    def execute(self, context, inputs):
+        """
+        :param flytekit.models.literals.LiteralMap inputs:
+        :rtype: dict[Text, flytekit.models.common.FlyteIdlEntity]
+        :returns: This function must return a dictionary mapping 'filenames' to Flyte Interface Entities.  These
+            entities will be used by the engine to pass data from node to node, populate metadata, etc. etc..  Each
+            engine will have different behavior.  For instance, the Flyte engine will upload the entities to a remote
+            working directory (with the names provided), which will in turn allow Flyte Propeller to push along the
+            workflow.  Where as local engine will merely feed the outputs directly into the next node.
+        """
+        inputs_dict = _type_helpers.unpack_literal_map_to_sdk_python_std(inputs, {
+            k: _type_helpers.get_sdk_type_from_literal_type(v.type) for k, v in _six.iteritems(self.interface.inputs)
+        })
+        outputs_dict = {
+            name: _task_output.OutputReference(_type_helpers.get_sdk_type_from_literal_type(variable.type))
+            for name, variable in _six.iteritems(self.interface.outputs)
+        }
+
+        inputs_dict.update(outputs_dict)
+
+        _exception_scopes.user_entry_point(self.task_function)(
+            _sdk_runnable.ExecutionParameters(
+                execution_date=context.execution_date,
+                execution_id=context.execution_id,
+                stats=context.stats,
+                logging=context.logging,
+                tmp_dir=context.working_directory
+            ),
+            **inputs_dict
+        )
+        return {
+            _constants.OUTPUT_FILE_NAME: _literal_models.LiteralMap(
+                literals={k: v.sdk_value for k, v in _six.iteritems(outputs_dict)}
+            )
+        }
+
+    def _get_container_definition(
+            self,
+            **kwargs
+    ):
+        """
+        :rtype: SdkRunnableSparkContainer
+        """
+        return super(SdkPyTorchTask, self)._get_container_definition(cls=SdkRunnablePytorchkContainer, **kwargs)
+
+    def _get_kwarg_inputs(self):
```

Comment:
Is this different than in the base SdkRunnableTask class?  Is it supposed to be?